### ISSUE TITLE
Remove zeroDataStates feature flag and unused code

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#5148]
## Feature Description

Once the zeroDataStates feature is complete and rolled out to 100% of eligible sites, we can safely remove the feature flag and remove the old code that no longer runs.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All usage of `zeroDataStates` feature flag should be removed, and any code that runs when `zeroDataStates === false` should be removed.

## Implementation Brief
* Remove `zeroDataStates` from `feature-flags.json`
* Search for the `zeroDataStates` string within the code base and review code where it is present. Any logic/component which was rendered conditionally when the `zeroDataStates` feature was enabled should be rendered unconditionally now. Any code which was rendered when the feature flag was disabled should be removed.
* Use of the `useFeature` hook with `zeroDataStates` as a parameter should be removed.
* Review stories where the `zeroDataStates` feature is enabled via the story's parameters to remove the parameter.
	* Legacy `Gathering data` stories must be removed.

### Test Coverage
* Update VRT images and E2E tests if necessary.

## QA Brief

* Set up a site with forced zero data for modules
* Ensure that this branch doesn't differ from the released plugin
* Ensure that no new warnings are raised in the console
* Check that the `zeroDataStates` feature flag is gone from the tester plugin as well

## Changelog entry

* Remove zeroDataStates feature flag and unused code.

**Comment by FlicHollis** [comment]
Hi @techanvil and @aaemnnosttv just checking if this is still blocked and that @techanvil is still planning on working on this?

**Comment by techanvil** [comment]
@FlicHollis this one is still blocked, pending the execution of 5047. I was still planning on working on the IB, as I did the IB for 5047 and the process will be something of a continuation, but someone else could do it too.

**Comment by FlicHollis** [comment]
That makes sense - thanks @techanvil

**Comment by mohitwp** [comment]
### QA Update ❌ 

@makiost 

I found few issues :

**Issue 1:**   Bracket ) character in the ‘All Traffic’ widget on develop branch. Evan confirmed that it is due to this PR. 

![image](https://user-images.githubusercontent.com/94359491/188116266-c0318b80-1ae4-494c-b3ff-8f8f0fc826c9.png)


**Issue 2:** Notices for gathering and zero state are no longer appearing on ` develop` branch.

<details>

https://user-images.githubusercontent.com/94359491/188116654-c7bfa24e-c9e0-430a-8100-ffbef164fd49.mp4


https://user-images.githubusercontent.com/94359491/188116703-0b1dc6ed-c7fe-410a-9b31-a959e299c585.mp4

</details>

**Comment by makiost** [comment]
@mohitwp Addressed the issues you pointed out in a new pr 👍

**Comment by aaemnnosttv** [comment]
@makiost is this ready for review again? I see it's still with you in Execution.

**Comment by makiost** [comment]
@aaemnnosttv All the fixes from QA are done, I just need to update one E2E test for it to be complete. Since the gathering data notifications are no longer hidden behind a feature flag, the notifications test fails, due to the gathering data notification being showed instead of the expected test notification.

**Comment by mohitwp** [comment]
### QA Update ✅ 

Verified

- Verified for the site which is in default gathering state.
- Verified for the site which is in forced gathering or zero state.
- Verified on main branch.
- On main branch 'Zero data state' feature flag not showing under tester plugin and zero data both state 'gathering and zero' working fine.
- No console errors.
- Also, CTA for both gathering and zero data state are now appearing.
- CTA appearing on both main,entity and WP dashboard.

<details>

![image](https://user-images.githubusercontent.com/94359491/188873794-a5495599-43d8-4004-9963-9c7d6a891c1b.png)

![image](https://user-images.githubusercontent.com/94359491/188873863-1a545b0e-2139-4e77-8702-526555fc38e1.png)

![image](https://user-images.githubusercontent.com/94359491/188873917-87d8ae7b-84f7-4c3a-b011-070feee918ab.png)

![image](https://user-images.githubusercontent.com/94359491/188874032-03b24383-cb65-44f7-bfde-b7f65459c1b6.png)

</details>
